packages/web-components/fast-foundation/src/picker/picker-list-item.ts

```diff
@@ -29,6 +29,17 @@ export class FASTPickerListItem extends FASTElement {
     @attr({ attribute: "value" })
     public value: string;
 
+    /**
+     * Disables the picker-list-item.
+     *
+     * @public
+     * @remarks
+     * HTML Attribute: disabled
+     */
+    @attr({ attribute: "disabled", mode: "boolean" })
+    public disabled: boolean;
```

Comment:
Is this needed on the "list item" primarily for styling? It feels like there should be a better way to handle this at least for known relationships. Like perhaps the list item listens to the parent picker's disable observable.

This feels like a similar issue to slider label needing to know the orientation. It's not the _label's_ orientation, but the parent. There's a comment there saying it will be removed some day, though I'm not sure how we'll properly style a slider label once it's gone.

Is there something we can add to fast-element to better support this?

Reply:
At the moment, it's also functional so the items don't emit events on click/keydown. However, we could just ignore events in the disabled parent.  Would we ever want to mix disabled/enabled items in a selection though?  ie. you can't delete the preselected item?

Reply:
I do think the item class should be able to modify js controlled behavior based on disabled or not.

Reply:
Maybe there's a dependency injection/context type solution where complex components like picker can publish some bindable state object that child components can easily access or bind to?

Reply:
I set up a picker context to pass this kind of state to children more easily.  Assuming it's the right way to go we could probably treat some other props this way.  Probably in data grid too.  Is this a good approach for component context  @EisenbergEffect ?